R/bs-theme.R

```diff
@@ -159,21 +159,22 @@ bs_theme_update <- function(theme, ..., preset = NULL, bg = NULL, fg = NULL,
                             font_scale = NULL, bootswatch = NULL) {
   assert_bs_theme(theme)
 
-  theme_has_preset <- any(grepl("^bs_(builtin|bootswatch)_", class(theme)))
-
   preset <- resolve_bs_preset(preset, bootswatch, version = theme_version(theme))
 
   if (!is.null(preset)) {
-    if (theme_has_preset) {
-      old_preset_class <- grep("^bs_(builtin|bootswatch)_", class(theme), value = TRUE)
-      old_preset_type <- sub("^bs_(builtin|bootswatch)_.+", "\\1", old_preset_class)
+    theme_has_preset <- inherits(theme, "bs_theme_with_preset")
```

Comment:
Could we use the value `$type` slot to represent presence of a preset (instead of a S3 class)?

Reply:
So the motivation for keeping all of the classes around was entirely about speed. Namely that it's a lot faster to check for the class than to look for the sass variables.

If we're not concerned about that (I mean it's faster but it's also not _that_ slow) then I'd be happy to drop all of the classes and just use `theme_preset_info()`.

There's also a secondary argument about future extensions that by providing a generic we can allow external expansions via new S3 methods. But I'm happy leaving that as a future bridge to cross.

Reply:
Ahh, great point, I'm fine on-board with keeping the classes then